cpp/src/arrow/util/async_util.cc

```diff
@@ -105,15 +105,16 @@ Status AsyncTaskGroup::AddTask(const Future<>& task_fut) {
   return AddTaskUnlocked(task_fut, std::move(guard));
 }
 
-Status AsyncTaskGroup::AddTaskIfNotEnded(const Future<>& task_fut) {
+Result<bool> AsyncTaskGroup::AddTaskIfNotEnded(const Future<>& task_fut) {
```

Comment:
I went ahead and added this for consistency but the return bool here can generally be ignored because the method doesn't actually start the work if it returns false.